### PR TITLE
Add schema to params for SourceTypes.

### PR DESCRIPTION
**Pull request by martinpovolny**
@agrare, @bdunne  : Do I need this to be able to access the `:schema` via the API?

### CODE REVIEW

File: app/controllers/api/v0/source_types_controller.rb [hunk 1]
```diff
@@ -12,11 +12,11 @@ def create
       private
 
       def create_params
-        body_params.permit(:name, :product_name, :vendor)
+        body_params.permit(:name, :product_name, :vendor, :schema)
```
Comment (bdunne) [inline app/controllers/api/v0/source_types_controller.rb]: :+1:

File: app/controllers/api/v0/source_types_controller.rb [hunk 2]
```diff
@@ -12,11 +12,11 @@ def create
...
       end
 
       def list_params
-        params.permit(:name, :product_name, :vendor)
+        params.permit(:name, :product_name, :vendor, :schema)
```
Comment (bdunne) [inline app/controllers/api/v0/source_types_controller.rb]: I don't think we'll need `:schema` here.  This is the params that can be passed to `model.where`, and I don't think people will search / filter by `:schema`.  I should probably come up with a better name for this method.  :smile:

**Comment by agrare** [comment]
> Do I need this to be able to access the :schema via the API?

For that I think you just need to add it to the [swagger yaml](https://github.com/ManageIQ/topological_inventory-api/blob/master/public/doc/swagger-2-v0.0.1.yaml#L1638-L1657)

File: app/controllers/api/v0/source_types_controller.rb (hunk 2 above)
Comment (martinpovolny) [reply app/controllers/api/v0/source_types_controller.rb]: fixing it...

**Comment by martinpovolny** [comment]
> For that I think you just need to add it to the swagger yaml

and run some command, or just add it?

**Comment by martinpovolny** [comment]
That is can I close this PR or do I actually need to do something?

**Comment by agrare** [comment]
> and run some command, or just add it?

Nope just add it, the serializer mixin looks to the swagger yaml to return permitted keys: https://github.com/ManageIQ/topological_inventory-api/blob/master/config/initializers/application_record_serializing.rb#L7

> That is can I close this PR or do I actually need to do something?

I think you can just add it to this PR since it goes together with adding `:schema` to the create params

File: public/doc/swagger-2-v0.0.1.yaml [hunk 3]
```diff
@@ -1655,6 +1655,8 @@ definitions:
       vendor:
         type: string
         example: Red Hat
+      schema:
+        type: string
```
Comment (agrare) [inline public/doc/swagger-2-v0.0.1.yaml]: @bdunne since schema is a jsonb type would `type: object` be more appropriate here or is string fine?

**Comment by martinpovolny** [comment]
Ok. Removed the `list_param` change and hand-edited the swagger yaml.

File: public/doc/swagger-2-v0.0.1.yaml (hunk 3 above)
Comment (bdunne) [reply public/doc/swagger-2-v0.0.1.yaml]: Yes, that's what I was thinking

**Comment by agrare** [comment]
@martinpovolny if you change schema type to object in the swagger yaml this looks good to me

**Pull request merged**